src/pages/Order/components/CartList.js

```diff
@@ -24,16 +25,15 @@ export default class CartList extends Component {
     const { productInCart } = this.state;
 
     const isAllChecked = productInCart.every(product => !!product.is_checked);
-
-    this.setState({ isAllChecked });
+    this.inputRef.current.checked = isAllChecked;
   };
 
   handleAllCheckedBox = e => {
     const { productInCart, isAllChecked } = this.state;
 
     productInCart.forEach(product => (product.is_checked = e.target.checked));

```

Comment:
```suggestion
    const newProductInCart = productInCart.map(product => {
      return { ...product, is_checked: e.target.checked }
    });
```
- productInCart라는 배열을 순회한 결과 다시 배열이 리턴되어야 하기 때문에 map을 사용해야 합니다
- product라는 객체에 접근해 내부의 값을 바꾼 뒤 새로운 객체를 리턴해야 하기 때문에 `{ ...product }`와 같이 새로운 객체 리터럴을 만들고 그 안에 값을 스프레드 해줘야 합니다
- 완성된 배열을 새로운 state에 담아 업데이트 해주면 불변성을 유지하며 업데이트하는 작업이 끝나게 됩니다.

아래는 map, reduce 등의 배열 메서드를 이해하는데 도움이 되었던 그림 자료인데 첨부해봅니다!

![스크린샷 2020-09-11 오전 3 27 49](https://user-images.githubusercontent.com/35240142/118581153-18aaf300-b7cc-11eb-8921-dd89b16062d8.png)

Reply:
감사합니다.

Reply:
제안주신 코드로는 기능 구현이 되지 않아 조금 다르게 구현되었습니다.